pkg/yurtctl/cmd/join/phases/postcheck.go

```diff
@@ -103,14 +103,21 @@ func checkYurthubHealthz() error {
 }
 
 //patchEdgeNode patch labels and annotations for edge-node.
-func patchEdgeNode(cfg *kubeadm.JoinConfiguration) error {
+func patchEdgeNode(cfg *kubeadm.JoinConfiguration, markAutonomous bool) error {
 	client, err := kubeconfigutil.ClientSetFromFile(kubeadmconstants.GetKubeletKubeConfigPath())
 	if err != nil {
 		return err
 	}
 	if err := patchnodephase.AnnotateCRISocket(client, cfg.NodeRegistration.Name, cfg.NodeRegistration.CRISocket); err != nil {
 		return err
 	}
+
+	if markAutonomous {
+		if err := apiclient.PatchNode(client, cfg.NodeRegistration.Name, annotateNodeWithAutonomousNode); err != nil {
+			return err
+		}
+	}
+
 	if err := apiclient.PatchNode(client, cfg.NodeRegistration.Name, func(n *v1.Node) {
 		n.Labels[projectinfo.GetEdgeWorkerLabelKey()] = "true"
```

Comment:
how about move `makeAutonomous` at here?

Reply:
Do you mean using `MarkAutonomousOptions` int `pkg/yurtctl/cmd/markautonomous/markautonomous.go`
```
	co := NewMarkAutonomousOptions()
        o.RunMarkAutonomous()
```
it's definitely better using this.

But, it has serval questions we have to answer:
1. a `kubeconf` file is required, and we may use `kubelet.conf`. Is this ok?
2.  join package import markautonomous package is ok but not advised I think.

Reply:
@adamzhoul I'm sorry that the comment is not clear. i mean that move the code line 114~120 after line 122, so reduce one node patch request.